lib/pages/chat_details/chat_details_page_view/same_type_events_list_builder.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+import 'package:fluffychat/pages/chat_details/chat_details_page_view/same_type_events_list_controller.dart';
+import 'package:fluffychat/widgets/twake_components/twake_smart_refresher.dart';
+
+class SameTypeEventsListBuilder extends StatelessWidget {
+  final SameTypeEventsListController controller;
+  final Future<Timeline> Function() getTimeline;
+  final Widget Function(BuildContext, Either<Failure, Success>) builder;
+
+  const SameTypeEventsListBuilder({
```

Comment:
What is the name? `SameType`?

Reply:
any suggestion for a generic builder class name for Media, Files, Links ?